clients/vault/src/horizon/client.rs

```diff
@@ -0,0 +1,254 @@
+use crate::{
+	error::Error,
+	horizon::{HorizonTransactionsResponse, Transaction},
+	oracle::ScpMessageHandler,
+};
+use async_trait::async_trait;
+use service::Error as ServiceError;
+use sp_std::{convert::From, str, vec::Vec};
+use std::{convert::TryInto, sync::Arc, time::Duration};
+use stellar::SecretKey;
+use stellar_relay::{sdk as stellar, sdk::Hash};
+use tokio::{sync::Mutex, time::sleep};
+
+const POLL_INTERVAL: u64 = 5000;
+
+pub const fn horizon_url(is_public_network: bool) -> &'static str {
+	if is_public_network {
+		"https://horizon.stellar.org"
+	} else {
+		"https://horizon-testnet.stellar.org"
+	}
+}
+
+// todo: temporary struct, as the real IssueRequest definition is in another branch.
+#[derive(Clone, PartialEq)]
+pub struct IssueRequest {
+	pub issue_id: Hash,
+}
+
+impl From<Vec<u8>> for IssueRequest {
+	fn from(val: Vec<u8>) -> Self {
+		IssueRequest { issue_id: val[0..31].try_into().unwrap() }
+	}
+}
+
+pub type IssueRequests = Vec<IssueRequest>;
+
+#[async_trait]
+pub trait HorizonClient {
+	async fn get_transactions(&self, url: &str) -> Result<HorizonTransactionsResponse, Error>;
+}
+
+#[async_trait]
+impl HorizonClient for reqwest::Client {
+	async fn get_transactions(&self, url: &str) -> Result<HorizonTransactionsResponse, Error> {
+		let response = self
+			.get(url)
+			.send()
+			.await
+			.map_err(|_| Error::HttpFetchingError)?
+			.json::<HorizonTransactionsResponse>()
+			.await
+			.map_err(|_| Error::HttpFetchingError)?;
+
+		Ok(response)
+	}
+}
+
+struct HorizonFetcher<C: HorizonClient> {
+	client: C,
+	vault_secret_key: String,
+	last_tx_id: Option<Vec<u8>>,
+}
+
+impl<C: HorizonClient> HorizonFetcher<C> {
+	pub fn new(client: C, vault_secret_key: String) -> Self {
+		Self { client, vault_secret_key, last_tx_id: None }
+	}
+
+	/// Fetch recent transactions from remote and deserialize to HorizonResponse
+	/// Since the limit in the request url is set to one it will always fetch just one
+	async fn fetch_latest_txs(
+		&self,
+		is_public_network: bool,
+	) -> Result<HorizonTransactionsResponse, Error> {
+		let vault_keypair: SecretKey = SecretKey::from_encoding(&self.vault_secret_key).unwrap();
+		let vault_address = vault_keypair.get_public();
+
+		let request_url = String::from(horizon_url(is_public_network)) +
+			"/accounts/" + str::from_utf8(vault_address.to_encoding().as_slice())
+			.map_err(|_| Error::HttpFetchingError)? +
+			"/transactions?order=desc&limit=1";
```

Comment:
The `limit=1` is problematic. We lived with it back in the days but we should tackle it now. We should fetch more than just the latest transaction because with this it might happen that we miss some.

I would suggest that, we do an initial fetch for the latest transaction, process that and somewhere store the `pagingToken` of that transaction. Now, for subsequent fetches, we can use the `pagingToken` (contained in the latest transaction we fetched) in the next calls, to be able to receive new transactions in batches. 
Info about pagination on stellar is [here](https://developers.stellar.org/api/introduction/pagination/).  But the relevant documentation on how to use a paging token when querying the transactions of an account is [here](https://developers.stellar.org/api/introduction/pagination/) where it is explained how to use it as a `cursor`.
You should use the pagingToken as cursor, set the order to "asc" and the limit to `20` or so.